tests/core/go_test/filter_test_cases_test.go

```diff
@@ -0,0 +1,144 @@
+package filter_test_cases_test
+
+import (
+	"encoding/xml"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: `
+-- BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
+
+go_test(
+    name = "skip_tests",
+    srcs = ["skip_tests_test.go"],
+	env = {"TESTBRIDGE_TEST_ONLY":"-^TestFoo$,-^TestBaz$"},
+)
+
+go_test(
+	name = "run_only",
+	srcs = ["run_only_test.go"],
+	env = {"TESTBRIDGE_TEST_ONLY":"^TestBar.+"},
+)
+
+go_test(
+	name = "filter_tests",
+	srcs = ["filter_tests_test.go"],
+	env = {"TESTBRIDGE_TEST_ONLY":"^TestTask.+,-^TestTaskB$"},
+)
+-- skip_tests_test.go --
+package filter_test_cases
+
+import "testing"
+
+func TestFoo(t *testing.T) {}
+func TestBar(t *testing.T) {}
+func TestBaz(t *testing.T) {}
+
+-- run_only_test.go --
+package filter_test_cases
+
+import "testing"
+
+func TestFooA(t *testing.T) {}
+func TestBarA(t *testing.T) {}
+func TestBarB(t *testing.T) {}
+
+-- filter_tests_test.go --
+package filter_test_cases
+
+import "testing"
+
+func TestTaskA(t *testing.T) {}
+func TestTaskB(t *testing.T) {}
+func TestTaskC(t *testing.T) {}
+func TestTaskD(t *testing.T) {}
+`,
+	})
+}
+
+// xml test suites to check which test cases were run
+type xmlTestCase struct {
+	XMLName xml.Name `xml:"testcase"`
+	Name    string   `xml:"name,attr"`
+}
+type xmlTestSuite struct {
+	XMLName   xml.Name      `xml:"testsuite"`
+	TestCases []xmlTestCase `xml:"testcase"`
+}
+type xmlTestSuites struct {
+	XMLName xml.Name       `xml:"testsuites"`
+	Suites  []xmlTestSuite `xml:"testsuite"`
+}
+
+func Test(t *testing.T) {
+	tests := []struct {
+		name                  string
+		args                  []string
+		expectedRunTestCases  map[string]bool
+		expectedSkipTestCases map[string]struct{}
+	}{
+		{
+			name:                  "skip_tests",
+			args:                  []string{"test", "//:skip_tests", "--test_env=GO_TEST_WRAP_TESTV=1"},
```

Comment:
Can you pass `--test_filter` here instead of setting `TESTBRIDGE_TEST_ONLY`? That way, you only need 1 `go_test` target to cover 3 cases.